addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/environment-sc-cidr-service.js

```diff
@@ -0,0 +1,221 @@
+  }
+
+  checkRequest(updateRequest) {
+    if (_.isUndefined(updateRequest) || _.isEmpty(updateRequest))
+      throw this.boom.badRequest('The request made had an empty body. Please check your payload', true);
+
+    const erroneousInputs = [];
+    const ipv6Format = [];
+    _.map(updateRequest, rule => {
+      _.forEach(rule.cidrBlocks, cidrBlock => {
+        if (IsCidr(cidrBlock) === 0) erroneousInputs.push(cidrBlock);
+        if (IsCidr.v6(cidrBlock)) ipv6Format.push(cidrBlock);
+      });
+    });
+
+    if (!_.isEmpty(erroneousInputs))
+      throw this.boom.badRequest(`The following are invalid CIDR inputs: ${erroneousInputs.join(', ')}`, true);
+
+    if (!_.isEmpty(ipv6Format))
+      throw this.boom.badRequest(
+        `Currently only IPv4 formatted CIDR inputs are allowed. Please remove : ${ipv6Format.join(
+          ', ',
+        )} CIDR ranges from your list`,
+        true,
+      );
+  }
+
+  async update(requestContext, { id, updateRequest }) {
+    // Before anything, check if the payload is valid
+    this.checkRequest(updateRequest);
+
+    const [environmentScService, lockService] = await this.service(['environmentScService', 'lockService']);
+    const existingEnvironment = await environmentScService.mustFind(requestContext, { id });
+
+    // Check if user is allowed to update cidrs
+    await this.assertAuthorized(
+      requestContext,
+      { action: 'update-sc', conditions: [this._allowAuthorized] },
+      { ...existingEnvironment, updateRequest },
+    );
+
+    await lockService.tryWriteLockAndRun({ id: `${id}-CidrUpdate` }, async () => {
+      // Calculate diff and update CIDR ranges in ingress rules
+      const { currentIngressRules, securityGroupId } = await environmentScService.getSecurityGroupDetails(
+        requestContext,
+        existingEnvironment,
+      );
+
+      // Perform the actual security group ingress rule updates
+      const newCidrList = await this.manageIngressRules(requestContext, {
+        existingEnvironment,
+        securityGroupId,
+        currentIngressRules,
+        updateRequest,
+      });
+
+      // Not storing the changes in the DB, but since all went well
+      // we return the cidr field as part of the env obj
+      existingEnvironment.cidr = newCidrList;
+    });
+
+    // Write audit event
+    await this.audit(requestContext, { action: 'update-environment-sc-cidr', body: existingEnvironment });
+
+    return existingEnvironment;
+  }
+
+  getIpPermission(updateRule, cidrList) {
+    const ipv4List = _.filter(cidrList, cidr => IsCidr.v4(cidr));
+    // const ipv6List = _.filter(cidrList, cidr => IsCidr.v6(cidr));
+
+    const IpPermission = {
+      FromPort: updateRule.fromPort,
+      IpProtocol: updateRule.protocol,
+      IpRanges: _.map(ipv4List, cidr => ({
+        CidrIp: cidr,
+        // For future: Custom description doesn't work, since that has to match exactly what's there in the SG.
+        // This would require storing cidr field in DB
```

Comment:
Isn't this comment misleading based on what we had discussed? We shouldn't require DB addition as long as we treat one IPAddress per rule.

Reply:
Yes. Had added that comment before our discussion. I'll update it.